### PR TITLE
Backfill resource claims after saturated pages

### PR DESCRIPTION
**Pull request by daniel-thom**
When claim_jobs_based_on_resources applies a client limit directly to the ready-job query, a worker can receive a page dominated by high-priority jobs that all require a saturated resource. In the common setup/GPU pipeline case, one GPU job can consume the available GPU while the remaining candidates in the limited page are also GPU jobs, leaving CPUs idle even though lower-priority CPU-only setup jobs are runnable.

Keep the client limit as the upper bound for returned jobs, but let the server make a bounded second query using the resources left after packing the primary page. This preserves priority ordering for the primary selection while allowing the server to backfill lower-priority work that fits the remaining CPU, memory, GPU, and node capacity.

Refactor the packing state into helper types so the primary and backfill passes share the same fit checks and job materialization path. Preserve the full priority/resource tie-break order from the existing scheduler behavior and keep the relaxed scheduler fallback consistent across both queries.

Enforce num_cpus > 0 on resource requirement create and update. Every real job must consume at least one CPU, and making that invariant explicit lets the backfill path short-circuit as soon as CPU capacity is exhausted instead of issuing a guaranteed-empty SQL query.

Add regression coverage for GPU-saturated backfill, relaxed scheduler fallback with backfill, exhausted-resource short-circuiting, and zero-CPU resource requirement rejection.

### CODE REVIEW

File: src/server/http_server/jobs_transport.rs [hunk 1]
```diff
@@ -1,6 +1,320 @@
...
+            separated.push_bind(job_id);
+        }
+        separated.push_unseparated(")");
+    }
+
+    builder.push(" ");
+    builder.push(RESOURCE_CLAIM_ORDER_BY);
+    builder.push(" LIMIT ");
+    builder.push_bind(remaining_limit as i64);
+
```
[the remark this answers is not in the document]
Comment (daniel-thom) [reply src/server/http_server/jobs_transport.rs:286]: Good point. This is a real remaining edge case, but it is separate from the GPU-saturation paging fix in this PR and needs its own bounded-fetch design/test coverage. I created https://github.com/NatLabRockies/torc/issues/267 to track it for a later PR.

**Review by nkeilbart — APPROVED** [review]
Looks good.
